Mathlib/Combinatorics/SimpleGraph/Copy.lean

```diff
@@ -54,18 +57,18 @@ abbrev Iso.toCopy (f : A ≃g B) : Copy A B := f.toEmbedding.toCopy
 
 namespace Copy
 
-/-- A copy gives rise to a homomorphism. -/
-abbrev toHom : Copy A B → A →g B := Subtype.val
+instance : FunLike (Copy A B) α β where
+  coe f := DFunLike.coe f.toHom
+  coe_injective' f g h := by obtain ⟨⟨_, _⟩, _⟩ := f; congr!
 
-@[simp] lemma coe_toHom (f : Copy A B) : ⇑f.toHom = f := rfl
+lemma injective (f : Copy A B) : Injective f.toHom := f.injective'
 
-lemma injective : (f : Copy A B) → (Function.Injective f.toHom) := Subtype.prop
+@[ext] lemma ext {f g : Copy A B} : (∀ a, f a = g a) → f = g := DFunLike.ext _ _
 
-instance : FunLike (Copy A B) α β where
-  coe f := DFunLike.coe f.toHom
-  coe_injective' _ _ h := Subtype.val_injective (DFunLike.coe_injective h)
+@[simp] lemma coe_toHom (f : Copy A B) : ⇑f.toHom = f := rfl
+@[simp] lemma toHom_apply (f : Copy A B) (a : α) : ⇑f.toHom a = f a := rfl
 
-@[simp] lemma coe_toHom_apply (f : Copy A B) (a : α) : ⇑f.toHom a = f a := rfl
```

Comment:
The new name is definitely better but don't forget to deprecate this (I guess it might break some of Mitchell's code)